### PR TITLE
Add unit tests for claws::box

### PR DESCRIPTION
**Pull request by doom**

### CODE REVIEW

File: modules/utils/claws/utils/box.hpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ namespace claws
     static_assert(!std::is_class_v<type>, "box should store a primitive, not a class");
 
   private:
-    type value{};
+    type _value{};
```
Comment (Gaspard--) [inline modules/utils/claws/utils/box.hpp]: Is there a specific reason for this renaming?

File: tests/claws-utils/box-test.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  static_assert(noexcept(int_box(std::declval<int_box &&>())));
+
+  static_assert(noexcept(std::declval<int_box &>() = std::declval<const int_box &>()));
+  static_assert(noexcept(std::declval<int_box &>() = std::declval<int_box &&>()));
+
+  static_assert(noexcept(int_box(std::declval<const int &>())));
+  static_assert(noexcept(int_box(std::declval<int &&>())));
+
+  static_assert(noexcept(static_cast<const int &>(std::declval<const int_box &>())));
+  static_assert(noexcept(static_cast<int &>(std::declval<int_box &>())));
```
Comment (Gaspard--) [inline tests/claws-utils/box-test.cpp]: Would be there a way to make a negative test? ex: check that claws::box doesn't compile with a class.

File: modules/utils/claws/utils/box.hpp (hunk 1 above)
Comment (doom) [reply modules/utils/claws/utils/box.hpp]: I tend to mark my private things with something, usually `_` as a prefix or suffix.

That also gets rid of the member shadowing in the custom constructors (which is not bad per se, but could trigger warnings under some configurations)

File: tests/claws-utils/box-test.cpp (hunk 2 above)
Comment (doom) [reply tests/claws-utils/box-test.cpp]: I don't see any way to achieve that here, as the static assertion in `claws::box` is a definition error, so SFINAE cannot be used to recover from it.

However, we could provide a `claws::is_boxable` type trait and change `claws::box` to use it, then test that trait.

Comment (Gaspard--) [reply tests/claws-utils/box-test.cpp]: I was wondering if such a thing would be present in google test. A trait isn't a bad idea, I'll open an issue.

File: modules/utils/claws/utils/box.hpp (hunk 1 above)
Comment (Gaspard--) [reply modules/utils/claws/utils/box.hpp]: I don't like the noise of `_`, and I find it more logical to use `this` when shadowing occurs.
`

**Pull request merged**